### ISSUE TITLE
Inherit from RestrictChildrenAttribute doesn't work.

### ISSUE THREAD

**Issue by hughzzq** [aspnet/Razor#1627]
Hi,

I have a sub class inherit from RestrictChildrenAttribute.  This class works correctly on ASP.NET Core 1.0/1.1.  But it does not on core 2.0.

**Comment by Eilon** [comment]
@hughzzq can you provide more details about what doesn't work? Can you share the structure or code of your tag helpers so that we can investigate?

**Comment by hughzzq** [comment]
@Eilon here's a very simple sample.  The expecte behavior of MyRestrictChildrenAttribute is if I type '<' inside a <testemail> tag, intelligence will should only show 'testemail-child'.  But it shows all tag.

    [MyRestrictChildrenAttribute("testemail-child")]
    //[RestrictChildrenAttribute("testemail-child")] //this works correctly.
    [HtmlTargetElement("testemail")]
    public class EmailTaghelper : TagHelper
    {
    }

    [HtmlTargetElement("testemail-child")]
    public class EmailChildTagHelper : TagHelper
    {
    }

    [AttributeUsage(AttributeTargets.Class, Inherited = false, AllowMultiple = false)]
    public class MyRestrictChildrenAttribute : RestrictChildrenAttribute
    {
        #region Ctors
        public MyRestrictChildrenAttribute(string tagName, params string[] tagNames)
            : base(tagName, tagNames)
        {
        }

        #endregion Ctors

    }
![incorrectbehavior](https://user-images.githubusercontent.com/29911584/29442226-d5580a6c-8402-11e7-8fa6-999345f7016d.png)
![expectbehavior](https://user-images.githubusercontent.com/29911584/29442234-df19c824-8402-11e7-9482-381d0882f090.png)

**Comment by Eilon** [comment]
@NTaylorMullen can you take a look? Is this expected?

**Comment by NTaylorMullen** [comment]
Interesting. In the old world we used to utilize `TypeInfo.GetCustomAttribute<RestrictChildren>(inherit: false)` which turns out does return sub-classed variants of the attributes. In 2.0.0 world we take a Roslyn approach to analyzing TagHelpers which does not take into account an attributes inheritance structure.

Honestly it's not a scenario we've put time into thinking about (hence why we don't have tests) but nevertheless it's something we should handle since we did before.
